core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -77,8 +77,9 @@ private Deletes() {
     return filter.filter(rows);
   }
 
-  public static StructLikeSet toEqualitySet(CloseableIterable<StructLike> eqDeletes, Types.StructType eqType) {
-    try (CloseableIterable<StructLike> deletes = eqDeletes) {
+  public static <T extends StructLike> StructLikeSet toEqualitySet(CloseableIterable<T> eqDeletes,
+                                                                   Types.StructType eqType) {
+    try (CloseableIterable<T> deletes = eqDeletes) {
```

Comment:
Why was this change needed?

Reply:
To avoid to transform `Record` to `StructLike` as your comment https://github.com/apache/iceberg/pull/3120#discussion_r711824192.

Reply:
Seems reasonable.